### PR TITLE
update guidlines

### LINKED ISSUE #15552: proper usage of :f>
:f> shortcut can be used to create function components from Reagent components (functions), where both RAtoms and Hooks work.

it seems like in some cases this works inefficient, investigate how to use it properly

it seems like when using reagent atoms from outside, functional component will be intialized again

like

```clojure
(fn []
  (let [atom (rf/sub [:sub]]
     (fn []
         [:f>
             (fn []
                [rn/text atom]
```

**SOLUTION:**

don't use anonymous inline functions

```clojure
(defn functional-comp []
[rn/text atom])


(fn []
  (let [atom (rf/sub [:sub]]
     (fn []
         [:f>  functional-comp])))           
```

### PR DESCRIPTION
**Pull request by flexsurfer**
fixes #15552

1) added part for inline functions usage in hiccup
2) relaxed styles rule
3) ~changed ### Always apply animated styles in the style file to Never
because there is no real reason to mix animated styles and hide animated function inside styles namespace , better to just simplify reanimated function~
4) don't mix props and styles in styles ns

### CODE REVIEW

File: doc/new-guidelines.md [hunk 1]
```diff
@@ -90,22 +133,27 @@ their styles in the top-level of the properties map, prefer to add them inside t
...
+;; good
 (defn circle
   []
   (let [opacity (reanimated/use-shared-value 1)]
     [reanimated/view {:style (reanimated/apply-animations-to-style
                               {:opacity opacity}
                               style/circle-container)}]))
 
-;; good
+;; bad
```
Comment (OmarBasem) [inline doc/new-guidelines.md]: I personally don't agree with this. Like that we would be mixing styles with view. Animation styles are still styles. There are animation styles that have 5+ style props, I don't think we should be having these in the view file.

Comment (flexsurfer) [reply doc/new-guidelines.md]: but its not like static styles right? its not about that you open styles ns and change style, its about defining which styles will be animated (dynamic) if you don't like map in your view you always can move it in a separate function with no problem, if you have 5+ style props it will look like `(style/circle-container something-prop1 something-prop2 something-prop3 something-prop4 something-prop5 )` so it doesn't look good anyway, and better to have somethingl like 

```clojure
(let [opacity (reanimated/use-shared-value 1)
       ra-styles  {:opacity opacity}]
    [ra/view (ra/apply-styles style/static ra-styles)])
```

**Review by Parveshdhull — APPROVED** [review]

File: doc/new-guidelines.md [hunk 2]
```diff
@@ -90,22 +133,49 @@ their styles in the top-level of the properties map, prefer to add them inside t
  ]
 ```
 
-### Always apply animated styles in the style file
+Also its fine to keep one liner styles in view
+
+```clojure
+;; ok
+[rn/view {:flex 1 :padding-top 5}]
```
Comment (ulisesmac) [inline doc/new-guidelines.md]: Please, add this style map inside a `:style` key in a map:

`{:style {:flex 1 :padding-top 5}}`

 just to make sure it's clear that we always want that `style` key.

**Review by ulisesmac — APPROVED** [review]

File: doc/new-guidelines.md (hunk 1 above)
Comment (ilmotta) [reply doc/new-guidelines.md]: At first I was on @flexsurfer's camp, then a comment from @briansztamfater convinced me to move to @OmarBasem camp, and I was cool there ;) I think styles in general don't look good anyway, but I tend to agree with Omar a little bit more.

File: doc/new-guidelines.md [hunk 3]
```diff
@@ -36,6 +36,49 @@ Pay special attention to:
...
+    (fn []
+      [:f>
+       (fn []
+         [rn/text atom]
+
+;; good
+(defn functional-comp [atom]
+ [rn/text atom])
+
+
```
Comment (ilmotta) [inline doc/new-guidelines.md]: Extra line

**Review by ilmotta — APPROVED** [review]
I don't agree with the changes proposed for reanimated styles, but if the rest of the crew is okay I'm fine with that as well.

File: doc/new-guidelines.md (hunk 1 above)
Comment (flexsurfer) [reply doc/new-guidelines.md]: could you pls share comment from @briansztamfater

Comment (flexsurfer) [reply doc/new-guidelines.md]: > I think styles in general don't look good anyway

could you elaborate?

Comment (ilmotta) [reply doc/new-guidelines.md]: > could you pls share comment from @briansztamfater

I would need a time machine @flexsurfer :laughing: The point in that PR thread was that I told Brian I preferred the animation code outside of the `style.cljs` file, since I preferred if functions inside the `style` ns are always pure (with the exception of functions dealing with colors). And I also mentioned `apply-animations-to-style` had a misleading name, since it's not applying anything like a normal function call, it's a hook with a special lifecycle, and I preferred (back then) that hooks are not used inside the `style` ns.

> I think styles in general don't look good anyway
>
> could you elaborate?

Bad wording on my part, sorry. I mean style maps are usually verbose and non-interesting. And the animation styles are also non-interesting and verbose (they pollute the view layer), so that's why I prefer if they're moved outside the `view` layer. This is why I agree with @OmarBasem and back then I agreed with @briansztamfater.

Comment (flexsurfer) [reply doc/new-guidelines.md]: thanks, I believe my suggestion reduces pollution as well but in a more proper way :)

Comment (ilmotta) [reply doc/new-guidelines.md]: I'm fine with your suggestion @flexsurfer, and thanks. I just hope we don't go back to this discussion again once the guidelines are updated and merged.

**Review by J-Son89 — APPROVED** [review]

File: doc/new-guidelines.md (hunk 1 above)
Comment (flexsurfer) [reply doc/new-guidelines.md]: i feel bad merging without @OmarBasem and @briansztamfater approves :(

File: doc/new-guidelines.md [hunk 4]
```diff
@@ -90,22 +133,49 @@ their styles in the top-level of the properties map, prefer to add them inside t
...
+ :right            0
+ :bottom           0}
+```
+
+
+### Never apply animated styles in the style file
+
+
+```clojure
+;; good
```
Comment (cammellos) [inline doc/new-guidelines.md]: `godd` -> `good`

File: doc/new-guidelines.md (hunk 1 above)
Comment (briansztamfater) [reply doc/new-guidelines.md]: @flexsurfer sorry I somehow skipped this PR, will check it now :)

Comment (briansztamfater) [reply doc/new-guidelines.md]: Yeah, so we discussed this some time ago, then the discussion leaded to update the guideline as it currently is. The link to the last time we defined this is https://github.com/status-im/status-mobile/pull/15489#discussion_r1149898021

In general terms, I agree with @OmarBasem and @ilmotta. One of the problems of not declaring animated styles in the styles file, is that it would make our components to have lot of boilerplate code for merging styles with animated styles, when we can abstract the responsibility to the style definition in the styles file and make our components look cleaner.
Also we could end up maintaining a same style for the same view on two different files, making it a bit more difficult to maintain and understand.
Regarding the naming, in the past I proposed renaming `apply-animations-to-style` to `merge-style`, the reason behind this is here https://github.com/status-im/status-mobile/pull/15275#issuecomment-1462182376. But I assume that is another topic

In conclusion I don't agree with this guideline change but I can adapt, the important thing IMO is to have a guideline to follow and make it immutable at least for a long time :P

File: doc/new-guidelines.md [hunk 5]
```diff
@@ -90,22 +133,49 @@ their styles in the top-level of the properties map, prefer to add them inside t
...
 
-### Always apply animated styles in the style file
+Also its fine to keep one liner styles in view
+
+```clojure
+;; ok
+[rn/view {:style {:flex 1 :padding-top 5}}]
+```
+
+### Don't define properties in styles ns
```
Comment (briansztamfater) [inline doc/new-guidelines.md:144]: Agree with this one

File: doc/new-guidelines.md (hunk 1 above)
Comment (briansztamfater) [reply doc/new-guidelines.md]: Also, regarding his example

> but its not like static styles right? its not about that you open styles ns and change style, its about defining which styles will be animated (dynamic) if you don't like map in your view you always can move it in a separate function with no problem, if you have 5+ style props it will look like `(style/circle-container something-prop1 something-prop2 something-prop3 something-prop4 something-prop5 )` so it doesn't look good anyway, and better to have somethingl like
> 
> ```clojure
> (let [opacity (reanimated/use-shared-value 1)
>        ra-styles  {:opacity opacity}]
>     [ra/view (ra/apply-styles style/static ra-styles)])
> ```

I feel that as more complex or big the animated style is, the more boilerplate the component will have just to set a style. And if we have many reanimated views in the same component, it might be worse.

 ```clojure
 (let [opacity (reanimated/use-shared-value 1)
       width (reanimated/use-shared-value 100)
       ra-styles {:opacity opacity
                  :width   width
                  :height  width}]
     [ra/view {:style (ra/apply-styles style/static ra-styles)}])
```
vs
 ```clojure
 (let [opacity (reanimated/use-shared-value 1)
       width (reanimated/use-shared-value 100)]
     [ra/view {:style (style/square opacity width)}])
```

So now let's add a new view to the component that moves horizontally

```clojure
 (let [opacity (reanimated/use-shared-value 1)
       width (reanimated/use-shared-value 100)
       translate-x (reanimated/use-shared-value 0)
       ra-styles-square {:opacity opacity
                         :width   width
                         :height  width}
       ra-styles-moving {:transform [{:translateX translate-x}]
                         :opacity   opacity}]
     [rn/view
       [ra/view {:style (ra/apply-styles style/static-square ra-styles-square)}]
       [ra/view {:style (ra/apply-styles style/static-moving ra-styles-moving)}]])
```
vs
 ```clojure
 (let [opacity (reanimated/use-shared-value 1)
       width (reanimated/use-shared-value 100)
       translate-x (reanimated/use-shared-value 0)]
     [rn/view
       [ra/view {:style (style/square opacity width)}]
       [ra/view {:style (style/moving opacity translate-x)}]])
```
It just feels more natural and understandable with both animated and static styles in the styles file, also a simpler guideline to follow for contributors.

Comment (ulisesmac) [reply doc/new-guidelines.md]: I don't like  the idea of having animated styles in the `style`  namespace, but it's true that doing this makes the view code more clean and easier to read, so I vote for having them in the `style` ns.

**Review by yqrashawn — APPROVED** [review]
